kernel/mm/phys_alloc.c

```diff
@@ -37,7 +37,7 @@ mem_block *phys_alloc_slab(uint64_t size, uint64_t align){
         if ((t_base_adj + size) <= (t_base + tmp->len)){
             return phys_split_block(tmp, t_base_adj, size);
         }
-    } while (tmp = tmp->next);
+    } while ((tmp = tmp->next));
```

Comment:
What's the compiler warning that this fixes?  GCC gives me nothing, and I can't see what semantic difference the extra set of parentheses would make (though it may be a case of it being so obvious I miss it, which I have been known to do).